test/WebJobs.Script.Tests/Configuration/ScriptJobHostOptionsSetupTests.cs

```diff
@@ -230,50 +196,44 @@ public void Configure_AppliesRetry_ExponentialBackOffDelay(string expectedStrate
             Assert.Equal(maxIntervalTimeSpan, options.Retry.MaximumInterval);
         }
 
-        [Fact]
-        public void Configure_TimeoutDefaultsNull_IfNotDynamic()
-        {
-            var options = GetConfiguredOptions(new Dictionary<string, string>());
-            Assert.Equal(ScriptJobHostOptionsSetup.DefaultFunctionTimeout, options.FunctionTimeout);
-
-            // When functionTimeout is set as null
-            var settings = new Dictionary<string, string>
-            {
-                { ConfigurationPath.Combine(ConfigurationSectionNames.JobHost, "functionTimeout"), string.Empty }
-            };
-
-            options = GetConfiguredOptions(settings);
-            Assert.Equal(ScriptJobHostOptionsSetup.DefaultFunctionTimeout, options.FunctionTimeout);
-        }
-
-        [Fact]
-        public void Configure_NoMaxTimeoutLimits_IfNotDynamic()
+        [Theory]
+        [InlineData(ScriptConstants.ElasticPremiumSku)]
+        [InlineData(ScriptConstants.FlexConsumptionSku)]
+        [InlineData("")]
+        public void Configure_NoMaxTimeoutLimits_ForSomeSkus(string sku)
         {
             var timeout = ScriptJobHostOptionsSetup.MaxFunctionTimeoutDynamic + TimeSpan.FromMinutes(10);
             string configPath = ConfigurationPath.Combine(ConfigurationSectionNames.JobHost, "functionTimeout");
             var settings = new Dictionary<string, string>
             {
                 { configPath, timeout.ToString() }
             };
+            var environment = new TestEnvironment();
+            environment.SetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteSku, sku);
 
-            var options = GetConfiguredOptions(settings);
+            var options = GetConfiguredOptions(settings, environment);
             Assert.Equal(timeout, options.FunctionTimeout);
         }
 
-        [Fact]
-        public void Configure_AppliesInfiniteTimeout_IfNotDynamic()
+        [Theory]
+        [InlineData(ScriptConstants.ElasticPremiumSku)]
+        [InlineData(ScriptConstants.FlexConsumptionSku)]
```

Comment:
Testing the other main change here - that FlexConsumption supports explicit specification of -1 in host.json to configure an unbounded timeout.